docs/maintenance/styleguide.md

```diff
@@ -49,9 +49,156 @@ Order buttons correctly:
 - ❌ Cancel (negative) | Save (positive)
 - ✅ Delete (negative) | ... (space) | Save (standard) | Publish (primary)
 
-### Captilization
+### Capitalization
 
 **Form labels**:
 
 - ✅ Resource type
 - ❌ Resource Type
+
+
+## Development practices
+
+### jQuery
+
+The most common usage of jQuery is when we apply JS to jinja (HTML) elements. 
+
+#### Selectors
+
+```html
+<div id="my-tab-menu" class="ui tab menu">
+  <div class="item active">Menu 1</div>
+  ...
+</div>
+```
+
+✅ DO
+
+```javascript
+$('#my-tab-menu').tab();
+```
+
+❌ DON'T
+```javascript
+$('.menu.tab').tab();
+```
+
+With the second approach you apply the function to all tab menu nodes on the page. What if there is a second menu? Or react element with the same class? None of your components/elements will behave as expected.  
+
+
+### Semantic UI
+
+#### Documentation
+
+Save your time and frustration, read the documentation (before you start with semantic UI)!
+
+Why this is important? You might rush into using available components without knowing their full power - you will find yourself "reinventing the wheel", instead of using features already available in semantic. 
+
+#### Say no to custom CSS (before it is too late)
+
+==Documentation first, custom CSS only as a last resort.==
+
+Semantic UI is quite complete as a theme, there is a possibility you can use one of the available props to style your component, instead of writing custom CSS (custom CSS is hard to maintain, especially for big projects, like ours).
+
+Example:
+
+✅ DO
+```javascript
+render() {
+  <Container>
+    <MyComponent/>
+  </Container>
+}
+```
+
+❌ DON'T
+
+```javascript
+render() {
+  <div className="my-centered-content"> // try to avoid custom styling!
+    <MyComponent/>
+  </div>
+}
+```
+
+```css
+.my-centered-content {
+  margin-left: auto;
+  margin-right: auto;
+}
+```
+
+#### Components
+
+Use dedicated react semantic ui components in react. The react version of semantic-UI components have normally implemented some logic, which is crucial for the component to work correctly. The plain CSS semantic UI might not have all the features available, therefore:  
```

Comment:
Comment: I see two styles used:

- Do, don't, why

vs 

- Why, Do, Don't

Perhaps just align on one, so that it follows same pattern.